myst_nb/execution.py

```diff
@@ -144,14 +149,17 @@ def generate_notebook_outputs(
 
             report_path = None
             if result.err:
-                report_path, message = _report_exec_fail(
-                    env,
-                    Path(file_path).name,
-                    result.exc_string,
-                    show_traceback,
-                    "Execution Failed with traceback saved in {}",
-                )
-                LOGGER.error(message)
+                if env.config["execution_fail_on_error"]:
+                    raise ExecutionError(str(result.err))
```

Comment:
do we have access to the name of the file being executed here? Perhaps the error string could be something like:

```
f"""
Execution failed for file: {file-name}
{str(result.err))
"""
```

(if it is already obvious from your implementation which file the error came from, then disregard this!)

Reply:
Great idea! I added the file path to the error message. The sphinx-build error now looks something like this:
```pytb
Exception occurred:
  File "/Users/jakevdp/github/executablebooks/MyST-NB/myst_nb/execution.py", line 153, in generate_notebook_outputs
    raise ExecutionError(
myst_nb.execution.ExecutionError: Execution failed for file: /Users/jakevdp/github/google/jax/docs/myst_nb_test.md
An error occurred while executing the following cell:
------------------
assert x[0] == 4321
------------------

---------------------------------------------------------------------------
AssertionError                            Traceback (most recent call last)
<ipython-input-3-381fab23ad2d> in <module>
----> 1 assert x[0] == 4321

AssertionError: 
AssertionError:
The full traceback has been saved in /var/folders/lr/h5hbphfs7_54btrvfk4qb7qc00h5r8/T/sphinx-err-1hoea1rl.log, if you want to report the issue to the developers.
Please also report this if it was a user error, so that a better error message can be provided next time.
A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks!
```